trace_viewer/core/find_controller.html

```diff
@@ -63,18 +64,38 @@
       this.filterHits_ = new tv.c.Selection();
       this.currentHitIndex_ = -1;
 
-      if (this.timeline_ && this.filterText.length) {
-        var filter = new tv.c.TitleFilter(this.filterText);
-        var filterHits = new tv.c.Selection();
-        var filterTask =
-            this.timeline.addAllObjectsMatchingFilterToSelectionAsTask(
-                filter, filterHits);
-        promise = Task.RunWhenIdle(filterTask);
-        promise.then(function() {
-          this.filterHitsDirty_ = false;
-          this.filterHits_ = filterHits;
-          this.timeline.setHighlightAndClearSelection(filterHits);
-        }.bind(this));
+      var navByFinderPattern = /^(\d+(\.\d+)?)@(\d+).(\d+)x(\d+(\.\d+)?)$/g;
+      var match = navByFinderPattern.exec(this.filterText);
+      if (match != null) {
+        var timestamp = match[1] * 1; // Convert string to a number.
+        var pid = match[3];
+        var tid = match[4];
+        var scaleX = match[5] * 1;
+
+        if (this.timeline_ && pid >= 0 && tid >= 0 && scaleX > 0) {
+          var loc = new tv.c.Location(this.timeline.viewport);
+          loc.fromStableIdAndTs(pid + '.' + tid, timestamp);
+
+          this.timeline.navToPosition(loc, scaleX);
+        }
+      } else {
+        if (this.filterText.length === 0) {
+          // Clear nav line and remove highlighted selection.
+          this.timeline.removeNavLine();
+          this.timeline.setHighlightAndClearSelection(new tv.c.Selection());
+        } else if (this.timeline_) {
+          var filter = new tv.c.TitleFilter(this.filterText);
+          var filterHits = new tv.c.Selection();
+          var filterTask =
+              this.timeline.addAllObjectsMatchingFilterToSelectionAsTask(
+                  filter, filterHits);
+          promise = Task.RunWhenIdle(filterTask);
+          promise.then(function() {
+            this.filterHitsDirty_ = false;
+            this.filterHits_ = filterHits;
+            this.timeline.setHighlightAndClearSelection(filterHits);
+          }.bind(this));
+        }
```

Comment:
Can we break this into 3 methods to make it a bit clearer?

if (match === null) {
  if (this.filterText.length ===) {
    clearNavigationMarker();
  } else if (this.timeline_) {
    find() ? (not sure what the righ tname is;
  }
} else {
  setNavigationMarker();
}

Something like that?